test/integration/targets/mongodb_replicaset/aliases

```diff
@@ -0,0 +1,6 @@
+destructive
+shippable/posix/group1
+skip/osx
+skip/freebsd
+needs/root
+needs/privileged
```

Comment:
This will prevent the tests from running on our CI system.

What error is it intended to address?

Reply:
Presumably you're referring to the skip entries? These have been there since day 1 and we've previously discussed why these are here. The new entry is only needs/privileged. This was added to see what effect it would have on shutdown times. I've been digging around on docker forums for similar problems and that was suggested. I didn't think it would have an effect as needs/root was already there (and I wasn't surprised to see it didn't).

Reply:
I was referring specifically to the `needs/privileged` alias, which will cause all the tests to be skipped by our CI system, since we don't run tests with privileged containers.

Reply:
OK. Isn't mentioned on the doc page https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/integration-aliases.html

They still run for needs/root though? needs/privileged can go if desired but if the shutdown times are such a problem why don't we add the disabled flag? Surely it would be a good idea to get some tests in that other contributors can manually run as part of their PRs for MongoDB modules. I think I've pretty much exhausted all ideas to solve this.

Reply:
I recommend updating the tests to support running on RHEL. Our CI system runs RHEL tests using dedicated VMs. That would eliminate any issues with docker interfering with the tests.

Reply:
Blaming docker sent me down a rabbit hole. This is not the cause of the long "shutdown" times as the same issue is experienced in VMs. The same behaviour is seen across all OS and OS Versions I have tried (quite a few). Perhaps this is something to do with the way Linux reports a port as drained. But a bit odd considering it's always a single mongod instance.  Using the pids module then checking for the process in /proc removed the long delays and prevented the port clashing. So that issue is resolved.

Support for CentOS / RHEL / Fedora has been implemented but I ended up excluding them because of a couple of intermittent errors. These do not happen locally at all and happened approx 50% on the time in CI.

remote_temp_dir sometimes fails with. All the tests actually work here and this is just a cleanup that happens at the end.

> "rmtree failed: [Errno 39] Directory not empty: '/tmp/ansible.13mfkh89.test/mongod3001'"

Ocassional MongoDB error. This happens after a restart of the replicaset. Not sure why. Pretty tricky for me to troubleshoot since I can't replicate it.

> pymongo.errors.ServerSelectionTimeoutError: localhost:3001: [Errno 111] Connection refused

For that reason I'd like to keep the exclusions in place for the moment.

Reply:
Nice work tracking that down.

The "directory not empty" error most likely means that another process is creating new files in the temporary directory while we're trying to delete it.

Reply:
Should be sorted. Added a one second delay to wait for task. I guess sometimes the OS keeps those file handles open a bit longer on occasion.